granulate_utils/linux/cgroups/base_cgroup.py

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+
+import os
+from enum import Enum
+from pathlib import Path
+from typing import List
+
+from granulate_utils.linux.cgroups.exceptions import MissingCgroup, MissingController, SkippedCgroup, UnsupportedCgroup
+
+PID_CGROUPS = Path("/proc/self/cgroup")
+CGROUP_PARENT_PATH = Path("/sys/fs/cgroup")
+
+
+class HIERARCHIES(Enum):
+    memory = "memory"
+    cpu = "cpu,cpuacct"
+
+
+class CgroupVerifications:
+    @staticmethod
+    def cgroup_v1() -> None:
+        if len(PID_CGROUPS.read_text().split("\n")) == 2:
+            raise UnsupportedCgroup("version 2")
+
+    @staticmethod
+    def supported_cgroup(cgroup_type: str) -> None:
+        assert cgroup_type, "must provide cgroup type"
```

Comment:
Not necessary? An empty string is not in `HIERARCHIES.__members__`, don't need this extra check